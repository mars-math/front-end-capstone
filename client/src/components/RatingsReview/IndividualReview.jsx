/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-useless-fragment */
/* eslint-disable react/no-array-index-key */
/* eslint-disable react/jsx-wrap-multilines */
/* eslint-disable react/jsx-indent */
/* eslint-disable react/button-has-type */
/* eslint-disable object-curly-spacing */
/* eslint-disable no-else-return */
/* eslint-disable import/extensions */
import React, { useState, useEffect } from 'react';
import axios from 'axios';

import ReviewImages from './ReviewImages.jsx';
import Stars from './ratingexampledata/stars.js';
import API_KEY from '../../../../config/config.js';

const {
  almostStar, quarterStar, halfStar, fullStar, emptyStar,
} = Stars;

export default function IndividualReview(props) {
  const {render} = props;
  const [helpfulNum, setHelpfulNum] = useState(render.helpfulness);
  const [buttonPressed, setButtonPressed] = useState(true);
  const [toggleBody, setToggleBody] = useState(false);

  axios.defaults.baseURL = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax';
  axios.defaults.headers.common.Authorization = API_KEY;

  // -----MESS WITH STATE FUNCTIONS-------------
  function incrementHelpfulness() {
    axios.put(`/reviews/${render.review_id}/helpful`)
      .then(() => setHelpfulNum((prevNum) => prevNum + 1))
      .catch((err) => console.log(err));
  }

  function bodyToggle() {
    setToggleBody(!toggleBody);
  }

  useEffect(() => {
    setHelpfulNum(render.helpfulness);
  }, [render]);
  // ------OTHER FUNCTIONS--------------
  function whichStar(rating, index) {
    if (rating >= index + 1) {
      return fullStar;
    } else {
      if (rating === (index + 0.5) || (rating > index + 0.33 && rating < index + 0.66)) {
        return halfStar;
      }
      if (rating <= index + 0.33 && rating > index) {
        return quarterStar;
      }
      if (rating >= index + 0.66) {
        return almostStar;
      }
      return emptyStar;
    }
  }

  function tooBig(body) {
    if (body.length > 250) {
      const shortened = `${body.slice(0, 250)}...`;
      return toggleBody ? <>
        <div style={{inlineSize: '550px', overflowWrap: 'break-word'}}>{body}</div>
       <button onClick={bodyToggle}>Show Less</button>
                          </>
        : <>
      <div style={{inlineSize: '550px', overflowWrap: 'break-word'}}>{shortened}</div>
      <button onClick={bodyToggle}>Show More</button>
          </>;
    } else {
      return <div style={{inlineSize: '550px', overflowWrap: 'break-word'}}>{body}</div>;
    }
  }

  function photosExist(photos) {
    if (photos.length !== 0) {
      return photos.map((photo, index) => <ReviewImages photo={photo.url} key={`photos${index}`} />);
    } else {
      return <></>;
    }
  }

  return (
      <div className="individual-review" style={{width: '92%'}}>
      {/* --------------STARS---------------- */}
      <div className="ind-top-row">
      <div>
      {[...Array(5)].map(
        // eslint-disable-next-line react/no-array-index-key
        (star, index) => <span key={`star${index}`}>{whichStar(render.rating, index)}</span>,
      )}
      </div>
      {/* --------------NAME---------------- */}
      <div style={{color: '#808080', fontSize: '.8em', marginLeft: 'auto'}}>
      {render.reviewer_name}
      -
      {' '}
      </div>
      {/* --------------DATE---------------- */}
      <div style={{color: '#808080', fontSize: '.8em'}}>{new Date(render.date).toLocaleString().split(',')[0]}</div>
      </div>
      {/* --------------summary---------------- */}
      <div style={{display: 'flex', width: '100%'}}>
        <h2 style={{margin: '8px'}}>{render.summary}</h2>
      </div>
      {/* --------------REVIEW BODY---------------- */}
      <div style={{display: 'flex', justifyContent: 'flex-start', margin: '5px', width: '100%'}}>
      {tooBig(render.body)}
      </div>
      {/* --------------RECOMMENDED?---------------- */}
      {render.recommend ? <div style={{marginLeft: '4px', marginBottom: '4px', fontWeight: 'bold'}}> âœ“ Recommended</div> : <></>}
      {/* --------------IS THERE SELLER RESPONSE?---------------- */}
      <div style={{display: 'flex', flexWrap: 'wrap', backgroundColor: '#C5C5C5'}}>
      {render.response ? (
        <>
          <h4 style={{margin: '8px 0 4px 0', width: '100%'}}>Response From the Seller: </h4>
          <div>{render.response}</div>
        </>
      ) : <></>}
      </div>
      {/* --------------ARE THERE PHOTOS?---------------- */}
      <div style={{
        display: 'flex', justifyContent: 'flex-start', margin: '5px', width: '100%',
      }}
      >
      {photosExist(render.photos)}
      </div>
      {/* --------------HELPFUL COUNTER---------------- */}
        <div style={{display: 'flex', justifyContent: 'flex-start', marginLeft: '4px'}}>
          <span>
          Helpful?
          {' '}
          </span>
        {buttonPressed && (
          <button
            style={{marginLeft: '5px', marginRight: '5px'}}
            className="star-filter-button"
            type="button"
            onClick={() => {
              incrementHelpfulness();
              setButtonPressed(!buttonPressed);
            }}
          >
          Yes
          </button>
        )}
        <span style={{color: '#808080', fontSize: '.8em'}}>
        (
        {helpfulNum}
        )
        </span>
        </div>
      </div>
  );
}
