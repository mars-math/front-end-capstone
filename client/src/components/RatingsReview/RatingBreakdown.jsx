/* eslint-disable react/prop-types */
/* eslint-disable import/extensions */
/* eslint-disable react/jsx-no-useless-fragment */
/* eslint-disable react/no-array-index-key */
/* eslint-disable array-callback-return */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import API_KEY from '../../../../config/config.js';
import Stars from './ratingexampledata/stars.js';
import ProgressBar from './ProgressBar.jsx';
import CharBreakdown from './CharBreakdown.jsx';

const {
  almostStar, quarterStar, halfStar, fullStar, emptyStar,
} = Stars;

export default function RatingBreakdown(props) {
  const [meta, setMeta] = useState({});

  const { manageFilter, url } = props;

  function getMeta(id) {
    return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/reviews/meta?product_id=${id}`, {
      headers: {
        Authorization: API_KEY,
      },
    })
      .then((data) => data.data)
      .catch(((err) => console.log(err)));
  }

  useEffect(() => {
    getMeta(url)
      .then((data) => {
        setMeta(data);
      })
      .catch((err) => console.log(err));
  }, []);

  function averageStars(ratingObj) {
    let total = 0;
    let count = 0;
    if (ratingObj) {
      Object.keys(ratingObj).map((key) => {
        total += Number(key) * ratingObj[key];
        count += Number(ratingObj[key]);
      });
    }
    return [(total / count).toFixed(1), count];
  }

  function whichStar(rating, index) {
    if (rating >= index + 1) {
      return fullStar;
    }
    if (rating === (index + 0.5) || (rating > index + 0.33 && rating < index + 0.66)) {
      return halfStar;
    }
    if (rating <= index + 0.33 && rating > index) {
      return quarterStar;
    }
    if (rating >= index + 0.66) {
      return almostStar;
    }
    return emptyStar;
  }

  return (
    <>
      <div className="total-stars-render" id="reviewsJump">
        <div style={{ fontSize: '3em' }}>{averageStars(meta.ratings)[0]}</div>
        {[...Array(5)].map(
          (star, index) => <span key={`star${index}`}>{whichStar(averageStars(meta.ratings)[0], index)}</span>,
        )}
      </div>

      <div style={{
        fontSize: '1.4em', fontWeight: 'bold', marginBottom: '10px', marginTop: '10px',
      }}
      >
        {averageStars(meta.ratings)[1]}
        {' '}
        reviews
      </div>

      {meta.ratings ? [1, 2, 3, 4, 5].map((item, idx) => (
        <ProgressBar
          key={`progress bar ${idx}`}
          bgcolor="#50C878"
          completed={(((meta.ratings[item]
            ? meta.ratings[item] : 0) / averageStars(meta.ratings)[1]) * 100).toFixed()}
          count={meta.ratings[item]}
          index={item}
          manageFilter={manageFilter}
        />
      )) : <></>}

      {meta.characteristics ? Object.keys(meta.characteristics).map((char, index) => (
        <div key={`charBreak ${index}`} style={{ marginBottom: '20px' }}>
          <h4 style={{ marginBottom: '5px', marginTop: '5px' }}>
            {char}
            :
          </h4>
          <CharBreakdown chars={meta.characteristics[char]} />
        </div>
      ))
        : <></>}
    </>
  );
}
