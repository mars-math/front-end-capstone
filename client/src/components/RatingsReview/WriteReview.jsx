import React, { useState, useEffect } from 'react';
import axios from 'axios';
import StarRating from './StarRating.jsx';
import API_KEY from '../../../../config/config.js';

function WriteReview(props) {
  const [isOpen, setIsOpen] = useState(false);
  const [charData, setcharData] = useState({});
  const [itemName, setItemName] = useState('');
  const [imgUrls, setImgUrls] = useState([]);
  const [imgUrl, setImgUrl] = useState('');
  const [postObj, setPostObj] = useState({});
  const [charObj, setCharObj] = useState({});

  const { updateRender, url } = props;

  axios.defaults.baseURL = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax';
  axios.defaults.headers.common.Authorization = API_KEY;

  function togglePopup() {
    setIsOpen(!isOpen);
  }

  useEffect(() => {
    axios.all([
      axios.get(`reviews/meta?product_id=${url}`),
      axios.get(`/products/${url}`),
    ])
      .then(axios.spread((data1, data2) => {
        setcharData(data1.data.characteristics);
        setItemName(data2.data.name);
      }));
  }, []);

  function buildCharObj(key, value) {
    const copyObj = { ...charObj };
    copyObj[key] = Number(value);
    setCharObj(copyObj);
  }

  function renderChars() {
    if (charData) {
      return (
        Object.keys(charData).map((char, index) => (
          <div key={`${char}${index}`} style={{ display: 'flex', flexWrap: 'wrap' }}>
            <div style={{ display: 'flex', width: '100%', fontWeight: 'bold' }}>
              {char}
              :
            </div>
            <fieldset style={{ display: 'flex', border: 'none' }} required onChange={(e) => buildCharObj(charData[char].id, e.target.value)}>
              <label htmlFor={char} className="char-selector">
                <input type="radio" id={char} name={char} value="1" />
                1
              </label>

              <label htmlFor={char} className="char-selector">
                <input type="radio" id={char} name={char} value="2" />
                2
              </label>

              <label htmlFor={char} className="char-selector">
                <input type="radio" id={char} name={char} value="3" />
                3
              </label>

              <label htmlFor={char} className="char-selector">
                <input type="radio" id={char} name={char} value="4" />
                4
              </label>

              <label htmlFor={char} className="char-selector">
                <input type="radio" id={char} name={char} value="5" />
                5
              </label>
            </fieldset>

          </div>
        ))

      );
    }
  }

  function imgButtonClick() {
    if (imgUrl.length > 5) {
      const imgUrlsCopy = [...imgUrls];
      imgUrlsCopy.push(imgUrl);
      setImgUrls(imgUrlsCopy);
    }
  }

  const buildPost = (key, value) => {
    const copyObj = { ...postObj };
    copyObj[key] = value;
    setPostObj(copyObj);
  };

  function postSubmit(e) {
    e.preventDefault();
    if (postObj.rating && (postObj.recommend !== undefined) && Object.keys(charObj).length === 4
    && postObj.body && postObj.name && postObj.email) {
      const realPostObj = { ...postObj };
      realPostObj.characteristics = charObj;
      realPostObj.product_id = url;
      realPostObj.photos = imgUrls;
      console.log(realPostObj);
      axios.post('/reviews', realPostObj)
        .then(() => {
          togglePopup();
          updateRender();
        })
        .catch((err) => console.log('Sadge', err));
    }
  }

  return (
    <>
      <button type="button" onClick={togglePopup} className="cust-button ">Write New Review</button>

      {isOpen && (
      <div className="popup-box">
        <div className="box">
          <span className="close-icon" onClick={togglePopup}>x</span>
          {/* --------TITLE----------- */}
          <h3>Write Your Review</h3>
          <h4 style={{ marginBottom: '20px', borderBottom: '2px solid black' }}>
            About the
            {' '}
            {itemName}
          </h4>
          <form onSubmit={(e) => postSubmit(e)}>
            {/* --------STAR RATING----------- */}
            <div style={{ marginBottom: '10px', fontWeight: 'bold' }}>Rate This Item:</div>
            <StarRating buildPost={buildPost} />
            {/* --------RECOMMENDED?----------- */}
            <div style={{ fontWeight: 'bold' }}>Recommend Product?</div>
            <fieldset style={{ border: 'none', marginBottom: '5px', borderBottom: '1px solid black'}} required onChange={(e) => buildPost('recommend', (e.target.value === 'yes'))}>
              <label htmlFor="yes">
                <input type="radio" id="yes" name="recommended" value="yes" />
                Yes
              </label>
              <br />
              <label htmlFor="no">
                <input type="radio" id="no" name="recommended" value="no" />
                No
              </label>
            </fieldset>
            {/* --------CHARACTERISTICS----------- */}
            {renderChars()}
            {/* --------SUMMARY----------- */}
            <label htmlFor="summary" className="write-review-labels" style={{borderTop: '1px solid black' }}>Summary:</label>
            <input
              style={{ width: '200px' }}
              type="text"
              name="summary"
              placeholder="Example: Best purchase ever!"
              onChange={(e) => buildPost('summary', e.target.value)}
            />
            {/* --------BODY----------- */}
            <label htmlFor="body" className="write-review-labels">Body:</label>
            <textarea
              placeholder="Why did you like the product or not?"
              name="body"
              cols="40"
              rows="5"
              required
              minLength="50"
              maxLength="1000"
              onChange={(e) => buildPost('body', e.target.value)}
            />
            <br />
            {/* --------IMG----------- */}
            {imgUrls.length > 0 ? imgUrls.map((img, index) => (
              <img
                key={`${img}${index}`}
                src={img}
                style={{ width: '50px', height: '50px', margin: '5px 10px 0 0' }}
                alt="thumbnail"
              />
            )) : <></>}
            <br />
            {imgUrls.length < 5
              ? (
                <>
                  <label htmlFor="img" className="write-review-labels">Img Url:</label>
                  <input type="text" name="img" onChange={(e) => setImgUrl(e.target.value)} />
                  <button type="button" onClick={imgButtonClick}>Upload</button>
                </>
              )
              : <></>}
            <br />
            {/* --------NICKNAME----------- */}
            <label htmlFor="nickname" className="write-review-labels">Nickname:</label>
            <input
              type="text"
              name="nickname"
              placeholder="Example: jackson11"
              required
              minLength="5"
              onChange={(e) => buildPost('name', e.target.value)}
            />
            <br />
            {/* --------EMAIL----------- */}
            <label htmlFor="email" className="write-review-labels">Email:</label>
            <input
              type="email"
              name="email"
              placeholder="Example: jackson11@email.com"
              required
              onChange={(e) => buildPost('email', e.target.value)}
            />
            <br />
            {/* --------SUBMIT BUTTON----------- */}
            <input type="submit" value="Write Review" className="cust-button" />
          </form>
        </div>
      </div>
      )}
    </>
  );
}

export default WriteReview;
