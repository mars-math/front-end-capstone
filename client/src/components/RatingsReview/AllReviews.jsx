/* eslint-disable import/extensions */
/* eslint-disable react/jsx-no-useless-fragment */
/* eslint-disable react/no-array-index-key */
import React, { useEffect, useState, useRef } from 'react';
import axios from 'axios';
import IndividualReview from './IndividualReview.jsx';
import RatingBreakdown from './RatingBreakdown.jsx';
import API_KEY from '../../../../config/config.js';
import WriteReview from './WriteReview.jsx';

export default function AllReviews() {
  const [amountToRender, setAmountToRender] = useState(2);
  const [renderedReviews, setRenderedReviews] = useState([]);
  const [filterStars, setFilterStars] = useState({});
  const slicedRender = renderedReviews.slice(0, amountToRender);
  const isFirstRender = useRef(true);

  function getReview(id, amount) {
    return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/reviews?count=${amount}&product_id=${id}`, {
      headers: {
        Authorization: API_KEY,
      },
    })
      .then((results) => results.data)
      .catch((err) => console.log(err));
  }

  const manageFilter = (star, currentState) => {
    if (!currentState) {
      setFilterStars({
        ...filterStars,
        [star]: star,
      });
    } else {
      const newFilter = { ...filterStars };
      delete newFilter[star];
      setFilterStars(newFilter);
    }
  };

  useEffect(() => {
    if (!isFirstRender.current) {
      getReview(43230, 999)
        .then((data) => {
          const sortedReviews = data.results.sort((a, b) => -a.date.localeCompare(b.date));
          setRenderedReviews(sortedReviews.filter((review) => {
            const arrStars = Object.keys(filterStars);
            if (arrStars.length === 0) {
              return true;
            }
            for (let i = 0; i < arrStars.length; i += 1) {
              if (review.rating === Number(arrStars[i])) {
                return true;
              }
            }
            return false;
          }));
        });
    } else {
      isFirstRender.current = false;
    }
  }, [filterStars]);

  const updateRender = () => {
    getReview(43230, 999)
      .then((data) => data.results.sort((a, b) => -a.date.localeCompare(b.date)))
      .then((results) => {
        setRenderedReviews(results);
      })
      .catch((err) => console.log(err));
  };

  useEffect(() => {
    updateRender();
  }, []);

  function sortReviews(e) {
    if (e.target.value === 'most recent ▼') {
      const renderByDate = [...renderedReviews];
      renderByDate.sort((a, b) => -a.date.localeCompare(b.date));
      setRenderedReviews(renderByDate);
    } else {
      const renderByHelpful = [...renderedReviews];
      renderByHelpful.sort((a, b) => b.helpfulness - a.helpfulness);
      setRenderedReviews(renderByHelpful);
    }
  }

  return (
    <>
      <div style={{
        display: 'flex', fontSize: '20px', justifyContent: 'center', marginLeft: '50px',
      }}
      >
        <div>{`${renderedReviews.length} reviews, sorted by`}</div>
        <select className="review-dropdown" onChange={(e) => sortReviews(e)}>
          <option value="most recent ▼">most recent</option>
          <option value="most helpful ▼">most helpful</option>
        </select>
      </div>
      <div className="all-review-components" data-testid="allRev-1">
        <div className="individual-reviews" style={{ maxHeight: '601px', overflow: 'auto', width: '45%' }}>
          {slicedRender.map((review, index) => (
            <div data-testid="tile">
              <IndividualReview render={review} key={`review${index}`} />
            </div>
          ))}
          {renderedReviews.length >= 2 && slicedRender.length < renderedReviews.length
            ? <button type="button" onClick={() => setAmountToRender((prevNum) => prevNum + 2)} className="cust-button">More Reviews</button> : <></>}
          <WriteReview updateRender={updateRender} />
        </div>
        <div className="rating-breakdown">
          <RatingBreakdown manageFilter={manageFilter} />
        </div>
      </div>
    </>
  );
}
