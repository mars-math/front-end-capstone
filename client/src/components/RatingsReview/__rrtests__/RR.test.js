import {
  render, screen, cleanup, act, waitFor,
} from '@testing-library/react';
import React from 'react';
// import axiosMock from 'axios';
import renderer from 'react-test-renderer';

import AllReviews from '../AllReviews.jsx';
import RatingBreakdown from '../RatingBreakdown.jsx';
import IndividualReview from '../IndividualReview.jsx';
import IndividualExample from '../ratingexampledata/individualExample.js';
import WriteReview from '../WriteReview.jsx';

afterEach(() => {
  cleanup();
});

test('should render AllReviews component', () => {
  render(<AllReviews />);
  const allReviewsElement = screen.getByTestId('allRev-1');
  expect(allReviewsElement).toBeInTheDocument();
  expect(allReviewsElement).toHaveTextContent('reviews');
});

xtest('shows 2 reviews initially', async () => {
  // axiosMock.get.mockResolvedValue(IndividualExample.All);

  // render(<AllReviews url="43230" />);
  // await waitFor(() => {
  //   expect(screen.getAllByTestId('tile')).toHaveLength(2);
  // });

  // const reviews = await waitFor(() => screen.getAllByTestId('tile'));

  // const res = await axiosMock.get('/products');

  // expect(reviews).toHaveLength(2);
  expect(res).toEqual(IndividualExample.All);
  // expect(screen.getByRole('button', { hidden: true })).not.toBeDisabled();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<AllReviews url="43230" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<RatingBreakdown url="43230" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<IndividualReview render={IndividualExample.justOne} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<WriteReview />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
