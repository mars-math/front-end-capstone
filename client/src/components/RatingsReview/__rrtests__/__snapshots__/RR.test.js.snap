// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "display": "flex",
        "fontSize": "20px",
        "justifyContent": "center",
        "marginLeft": "50px",
      }
    }
  >
    <div>
      0 reviews, sorted by
    </div>
    <select
      className="review-dropdown"
      onChange={[Function]}
    >
      <option
        value="most recent ▼"
      >
        most recent
      </option>
      <option
        value="most helpful ▼"
      >
        most helpful
      </option>
    </select>
  </div>,
  <div
    className="all-review-components"
    data-testid="allRev-1"
  >
    <div
      className="individual-reviews"
      style={
        Object {
          "maxHeight": "601px",
          "overflow": "auto",
          "width": "45%",
        }
      }
    >
      <button
        className="cust-button "
        onClick={[Function]}
        type="button"
      >
        Write New Review
      </button>
    </div>
    <div
      className="rating-breakdown"
    >
      <div
        className="total-stars-render"
      >
        <div
          style={
            Object {
              "fontSize": "3em",
            }
          }
        >
          NaN
        </div>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
      </div>
      <div
        style={
          Object {
            "fontSize": "1.4em",
            "fontWeight": "bold",
            "marginBottom": "10px",
            "marginTop": "10px",
          }
        }
      >
        0
         
        reviews
      </div>
    </div>
  </div>,
]
`;
